### PR TITLE
Add ability to switch octree implementations.

### LINKED ISSUE #1416: Add option to switch octree impls in an existing scene.
I believe octree has a switch impl method, or we can serialize the octree to a temporary file and deserialize it with the alternate implementation.

### PR DESCRIPTION
**Pull request by ThatRedox**
When complete closes #1416

Complete:
- Switch octrees in an existing scene
- Progress bar while switching octrees

To do:
- [x] Check it saves the new octree setting
- [x] Block rendering when switching octrees (can cause NPE)

### CODE REVIEW

**Comment by ThatRedox** [comment]
Octree is now set to an empty octree when switching.

**Comment by jackjt8** [comment]
If using an octree provided by the Octree Plugin this does not work but no error is thrown. 

image

---

Otherwise, switching between built in Octrees works as expected.

**Comment by jackjt8** [comment]
Unfortunately, it does not write the updated octree or update the .json so you are stuck with the old octree.

**Comment by ThatRedox** [comment]
Added soft refresh to propagate octree (and bvh) changes without forcing a rerender. Closes #1005. Closes #955.

**Comment by ThatRedox** [comment]
> If using an octree provided by the Octree Plugin this does not work but no error is thrown.
> 
> image
> 
> Otherwise, switching between built in Octrees works as expected.

The progress overflow should be fixed. I can't seem to reproduce the failing with no error thrown (perhaps it was working but the progress bar just overflowed).

In general, I have found that `NODE` appears to be completely broken for me (#1434) and `DICTIONARY` is missing blocks when loading from a file so they may not be the best to test with.

**Comment by jackjt8** [comment]
> Added soft refresh to propagate octree (and bvh) changes without forcing a rerender. Closes #1005. Closes #955.

Can confirm that this is working as intended.

> The progress overflow should be fixed. I can't seem to reproduce the failing with no error thrown (perhaps it was working but the progress bar just overflowed).

At the very least the fix is working.

> ... `DICTIONARY` is missing blocks when loading from a file ([issue](https://github.com/aTom3333/chunky-octree-plugin/issues/2)) so they may not be the best to test with.

I will investigate this issue further.

**Comment by jackjt8** [comment]
This PR is ready last time I tested it. It would just not work correctly with the octree plugin.

**Comment by aTom3333** [comment]
Apart from the `DICTIONARY` octree issue, was there something else preventing this to work with the octree plugin?

**Comment by jackjt8** [comment]
> Apart from the `DICTIONARY` octree issue, was there something else preventing this to work with the octree plugin?

Actually no. I just tested the PR further after all the fixes and technically there is no issue with the Plugin and this PR.

File: chunky/src/java/se/llbit/chunky/renderer/DefaultRenderManager.java [hunk 1]
```diff
@@ -246,6 +246,11 @@ public void run() {
         // Wait for a scene state change (e.g. user editing the scene)
         ResetReason reason = sceneProvider.awaitSceneStateChange();
 
+        if (reason == ResetReason.SETTINGS_CHANGED_SOFT) {
+          // Settings sync reset. No need to re-render.
+          continue;
+        }
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/renderer/DefaultRenderManager.java:252]: How about introducing a new `requiresRerender` to the `ResetReason` and then

```suggestion
        if (!reason.isRerenderRequired()) {
          // Settings sync reset. No need to re-render.
          continue;
        }
```

**Review by leMaik — APPROVED** [review]
lgtm. We could refactor the soft-reset render reason, but it's not a blocker.

**Review by leMaik — APPROVED** [review]

**Pull request merged**